### PR TITLE
Formatting for the group multiline alignment for match arm arrows #6074

### PR DESCRIPTION
**Pull request by junichi11**
- https://github.com/apache/netbeans/issues/6074
- Add "Match Arm Arrow" as an alignment option
- Add/Fix unit tests

Example:
```php
match ($type1) {
    'integer', 'test' => 'test',
    'string' => match ($type2) {
        1, 2 => "test",
        3 => "test",
        4 => "test",
    },
    default => null,
};
```

Result:

```php
match ($type1) {
    'integer', 'test' => 'test',
    'string'          => match ($type2) {
        1, 2 => "test",
        3    => "test",
        4    => "test",
    },
    default           => null,
};
```
nb-php-gh-6074-match-arm-arrow-alignment-option

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/indent/FormatToken.java [hunk 1]
```diff
@@ -315,22 +321,28 @@ public static class AssignmentAnchorToken extends FormatToken {
...
             length = -1;
             maxLength = -1;
             previous = null;
             isInGroup = false;
             this.multilined = multilined;
+            this.type = type;
         }
 
-        public int getLenght() {
+        public int getLength() {
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/indent/FormatToken.java:340]: Fixed typo.

File: php/php.editor/src/org/netbeans/modules/php/editor/indent/FormatToken.java [hunk 2]
```diff
@@ -315,22 +321,28 @@ public static class AssignmentAnchorToken extends FormatToken {
...
             return length;
         }
 
-        public void setLenght(int lenght) {
-            this.length = lenght;
+        public void setLength(int length) {
+            this.length = length;
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/indent/FormatToken.java:345]: Fixed typo.

File: php/php.editor/src/org/netbeans/modules/php/editor/indent/TokenFormatter.java [hunk 3]
```diff
@@ -3211,53 +3244,71 @@ private int countSpacesForGroupedToken(final FormatToken.AssignmentAnchorToken t
 
         private int countSpacesWhenNotExpandingTabs(final FormatToken.AssignmentAnchorToken token) {
             int spaces;
+            int spaceAroundAssignment = getSpaceAroundAssignment(token);
+            // consider also the space around assinment here
+            // to avoid adding extra spaces before assignment
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/indent/TokenFormatter.java:3249]: I've changed this method behavior.

e.g. See below unit tests:
- https://github.com/apache/netbeans/pull/6884/files#diff-bcbc4870ac3ca73b9a4a6ff60602fc997de403e1ff13f370a5652d9a340f96fb
- https://github.com/apache/netbeans/pull/6884/files#diff-08ebd15b297e731302f5c9374c2f57687a0d58f0b41b1cdca4ac40b014931742

Before:

nb-php-gh-6074-match-arm-arrow-alignment-extra-space-before

After:

nb-php-gh-6074-match-arm-arrow-alignment-extra-space-after

File: php/php.editor/test/unit/data/testfiles/formatting/alignment/issue210617.php.formatted [hunk 4]
```diff
@@ -16,35 +16,35 @@ $var111111	 = 'test';
...
+$v			= 'test';
+$va			= 'test';
+$var		= 'test';
+$var1		= 'test';
+$var11		= 'test';
+$var111		= 'test';
+$var1111	= 'test';
+$var11111	= 'test';
+$var111111	= 'test';
+$var1111111 = 'test';
```
Comment (junichi11) [inline php/php.editor/test/unit/data/testfiles/formatting/alignment/issue210617.php.formatted:28]: Tab space is 4 in this test.

nb-php-gh-6884-test-result-tab4spaces

File: php/php.editor/src/org/netbeans/modules/php/editor/indent/TokenFormatter.java (hunk 3 above)
Comment (tmysik) [reply php/php.editor/src/org/netbeans/modules/php/editor/indent/TokenFormatter.java:3249]: Looks correct to me now 👍

File: php/php.editor/src/org/netbeans/modules/php/editor/indent/CodeStyle.java [hunk 5]
```diff
@@ -584,6 +584,10 @@ public boolean groupMulitlineArrayInit() {
         return preferences.getBoolean(GROUP_ALIGNMENT_ARRAY_INIT, getDefaultAsBoolean(GROUP_ALIGNMENT_ARRAY_INIT));
     }
 
+    public boolean groupMulitlineMatchArmArrow() {
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/indent/CodeStyle.java:587]: Typo, Mul[t]i[t]line

**Review by tmysik — APPROVED** [review]
Great to have all the tests, thanks!

File: php/php.editor/src/org/netbeans/modules/php/editor/indent/CodeStyle.java (hunk 5 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/indent/CodeStyle.java:587]: https://github.com/apache/netbeans/blob/e8b206ef646754497a6890a1aa44ea2aa4cfcd8a/php/php.editor/src/org/netbeans/modules/php/editor/indent/CodeStyle.java#L579-L585

Thanks for catching it :) I've copied it from the above method. I'll also fix them.

**Comment by junichi11** [comment]
@tmysik Thank you for your quick review! I'll merge it after I fix the typos :)

File: php/php.editor/src/org/netbeans/modules/php/editor/indent/CodeStyle.java (hunk 5 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/indent/CodeStyle.java]: Fixed: https://github.com/apache/netbeans/compare/5271e2901f9a8c7421e518a0f5c9f8dc4ca7ea20..53dd6d320a66f0521e22e77fdb6f24a4c45eb30d

**Pull request merged**